tests/unit/test_backend_dist.py

```diff
@@ -0,0 +1,275 @@
+import PyRDF, unittest, ROOT
+from PyRDF import backend, Spark
+
+class SelectionTest(unittest.TestCase):
+    """
+    A series of tests to check the
+    accuracy of 'PyRDF.use' method.
+
+    """
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Cleans up the `SparkContext` objects
+        that were created during the tests in
+        this class.
+
+        """
+        from pyspark import SparkContext
+        context = SparkContext.getOrCreate()
+        context.stop()
+
+    def test_spark_select(self):
+        """
+        Test to check if 'spark'
+        environment gets set correctly.
+
+        """
+
+        PyRDF.use("spark")
+        self.assertIsInstance(PyRDF.current_backend, Spark)
+
+class DistBackendInitTest(unittest.TestCase):
+    """
+    Tests to ensure that the Dist abstract
+    class cannot be instantiated.
+
+    """
+    def test_dist_init_error(self):
+        """
+        Test case to check that any attempt to
+        instantiate the `Dist` abstract class
+        results in a `TypeError`.
+
+        """
+        with self.assertRaises(TypeError):
+            backend.Dist()
+
+    def test_subclass_without_method_error(self):
+        """
+        Test case to check that creation of a
+        subclass without implementing `processAndMerge`
+        method throws a `TypeError`.
+
+        """
+        class TestBackend(backend.Dist):
+            pass
+
+        with self.assertRaises(TypeError):
+            TestBackend()
+
+class SparkBackendInitTest(unittest.TestCase):
```

Comment:
I'm not understanding the way you divided the tests. Didn't we agree that you should have separate files for Spark, Dist and Common? I'm seeing Spark and Dist together...

Reply:
And if this is Spark related, it should have spark in the filename.

Reply:
@diocas , I think we decided to split them by `dist`, `local` and `common`.  Am I right @etejedor ?

Reply:
Spark should also have its own file. Proposal:
- `PyRDF/tests/backend`
- `PyRDF/tests/backend/test_common.py`
- `PyRDF/tests/backend/test_local.py`  (if we need it)
- `PyRDF/tests/backend/test_dist.py`  (if we need it)
- `PyRDF/tests/backend/test_spark.py`

Reply:
Will fix this @etejedor :+1: